File: solutions/binary-tree-inorder-traversal/binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

// check if input is empty
    // if so then return empty array
// create arr to store values
// check if root has left child
    // if so then recurse with left child and get array value
// push current value of root into array
// check if root has right child
    // if so then recurse with right child and get array value
// return array

var inorderTraversal = function(root) {
    if (!root) {
       return [];
    }
    
    let values = [];
    
    if (root.left) {
        values = values.concat(inorderTraversal(root.left));
    }
    
    values.push(root.val);
    
    if (root.right) {
        values = values.concat(inorderTraversal(root.right));
    }
        
    return values;
};

File: solutions/binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# check if input is empty
#     if so then return empty array
# create arr to store values
# check if root has left child
#     if so then recurse with left child and get array value
# push current value of root into array
# check if root has right child
#     if so then recurse with right child and get array value
# return array

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []
        
        values = []
        
        if root.left is not None:
            values = values + self.inorderTraversal(root.left)
        
        values.append(root.val)
        
        if root.right is not None:
            values = values + self.inorderTraversal(root.right)
        
        return values
